src/main/java/com/io/image/manager/service/operations/CropOperation.java

```diff
@@ -11,7 +11,42 @@ public CropOperation(String name, Map<String, String> arguments) {
 
     @Override
     public BufferedImage run(BufferedImage image) throws ImageOperationException {
-        //TODO remember to handle errors properly (throw ImageOperationException when arg is wrong)
-        return image;
+
+        int startXPoint = Integer.parseInt(getArguments().getOrDefault("x", Integer.toString(0)));
+        int startYPoint = Integer.parseInt(getArguments().getOrDefault("y", Integer.toString(0)));
+
+        prepareNewDimensionImage(image.getWidth(), image.getHeight(), startXPoint, startYPoint);
+        checkValidationOfArguments(startXPoint, startYPoint, image.getWidth(), image.getHeight());
+
+        return image.getSubimage(startXPoint, startYPoint, newWidth, newHeight);
+    }
+
+    public void prepareNewDimensionImage(int imageWidth, int imageHeight, int startXPoint, int startYPoint) {
+        newWidth = Integer.parseInt(getArguments().getOrDefault("w", Integer.toString(imageWidth - startXPoint)));
+        newHeight = Integer.parseInt(getArguments().getOrDefault("h", Integer.toString(imageHeight - startYPoint)));
+    }
+
+    public void checkValidationOfArguments(int startXPoint, int startYPoint, int imageWidth, int imageHeight)
+            throws ImageOperationException {
+
+        // origin point is outside of the original image bounds
+        if (startXPoint < 0 || startYPoint < 0 || startXPoint > imageWidth || startYPoint > imageHeight) {
+            throw new ImageOperationException();
```

Comment:
I think that in documentation we had to give the reason of failed operation (I might be wrong though). Can we pass the reason to the exception and then return it as a message in json returned with 400 response?